core/src/main/java/nu/yona/server/messaging/service/MessageService.java

```diff
@@ -285,16 +295,45 @@ private DtoManager getManager(Message messageEntity)
 	}
 
 	@Transactional
-	public void sendMessageAndFlushToDatabase(Message message, MessageDestinationDto destination)
+	public void sendMessageAndFlushToDatabase(Message message, UserAnonymizedDto toUser)
 	{
-		MessageDestination destinationEntity = getMessageDestination(destination.getId());
+		sendMessage(message, toUser);
+
+		MessageDestination destinationEntity = getMessageDestination(toUser.getAnonymousDestination().getId());
+		messageDestinationRepository.saveAndFlush(destinationEntity);
+	}
+
+	@Transactional
+	public void sendNamedMessageAndFlushToDatabase(Message message, User toUser)
```

Comment:
In other places, this is called a "direct message". That sounds better to me, as the above name could suggest the message has a name, while it's actually the message destination that has one.